airflow/providers/google/cloud/hooks/pubsub.py

```diff
@@ -545,8 +549,22 @@ def acknowledge(
         :param metadata: (Optional) Additional metadata that is provided to the method.
         :type metadata: Sequence[Tuple[str, str]]]
         """
+
         if not project_id:
             raise ValueError("Project ID should be set.")
+
+        if ack_ids is not None and messages is not None:
+            raise ValueError("'ack_ids' and 'messages' arguments are mutually exclusive.")
+
+        if ack_ids is None:
+            if messages is None:
+                raise ValueError("Either 'ack_ids' or 'messages' argument have to be provided.")
```

Comment:
I think this case is handled in 556. At this point we are sure that `ack_ids` or `messages` is not None/

Reply:
At 556, we assure that both are not provided at the same time.
Here we check that at least one of them is provided.
Together these two checks assure that one and only one of them is provided.

Reply:
An equivalent if-tree would be:

```py
        if ack_ids is not None and messages is None:
            pass
        elif ack_ids is None and messages is not None:
            ack_ids = [
                message.ack_id
                for message in messages
            ]
        else:
            raise ValueError("One and only one of 'ack_ids' and 'messages' arguments have to be provided")
```

If you consider it more readable than current one, I can change it.

Reply:
+1 for the if-tree

Reply:
Done